client/core/core.go

```diff
@@ -7046,3 +7047,140 @@ func (c *Core) DeleteArchivedRecords(olderThan *time.Time, matchesFile, ordersFi
+	if err != nil {
+		return "", fmt.Errorf("Accelerate password error: %w", err)
+	}
+
+	tracker, swapCoinIDs, accelerationCoins, changeCoinID, requiredForRemainingSwaps, err := c.orderAccelerationParameters(oidB)
+	if err != nil {
+		return "", err
+	}
+
+	newChangeCoin, txID, err :=
+		tracker.wallets.fromWallet.accelerateOrder(swapCoinIDs, accelerationCoins, changeCoinID, requiredForRemainingSwaps, newFeeRate)
+	if err != nil {
+		return "", err
+	}
+
+	tracker.metaData.ChangeCoin = order.CoinID(newChangeCoin.ID())
+	tracker.metaData.AccelerationCoins = append(tracker.metaData.AccelerationCoins, tracker.metaData.ChangeCoin)
+	tracker.coins[newChangeCoin.ID().String()] = newChangeCoin
+	err = tracker.db.UpdateOrder(tracker.metaOrder())
+	if err != nil {
+		c.log.Errorf("AccelerateOrder: failed to update order in database: %v", err)
+	}
+
+	return txID, nil
+}
+
+// AccelerationEstimate returns the amount of funds that would be needed to
+// accelerate the swap transactions in an order to a desired fee rate.
+func (c *Core) AccelerationEstimate(oidB dex.Bytes, newFeeRate uint64) (uint64, error) {
+	tracker, swapCoins, accelerationCoins, changeCoin, requiredForRemainingSwaps, err := c.orderAccelerationParameters(oidB)
+	if err != nil {
+		return 0, err
+	}
+
+	accelerationFee, err := tracker.wallets.fromWallet.accelerationEstimate(swapCoins, accelerationCoins, changeCoin, requiredForRemainingSwaps, newFeeRate)
+	if err != nil {
+		return 0, err
+	}
+
+	return accelerationFee, nil
+}
+
+// PreAccelerateOrder returns information the user can use to decide how much
+// to accelerate stuck swap transactions in an order.
+func (c *Core) PreAccelerateOrder(oidB dex.Bytes) (*PreAccelerate, error) {
+	tracker, swapCoinIDs, accelerationCoins, changeCoinID, requiredForRemainingSwaps, err := c.orderAccelerationParameters(oidB)
+	if err != nil {
+		return nil, err
+	}
+
+	feeSuggestion := c.feeSuggestionAny(tracker.fromAssetID)
+
+	currentRate, suggestedRange, err :=
+		tracker.wallets.fromWallet.preAccelerate(swapCoinIDs, accelerationCoins, changeCoinID, requiredForRemainingSwaps, feeSuggestion)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PreAccelerate{
+		SwapRate:       currentRate,
+		SuggestedRate:  feeSuggestion,
+		SuggestedRange: suggestedRange,
+	}, nil
+}
+
+// orderAccelerationParameters takes an order id, and returns the parameters
+// needed to accelerate the swap transactions in that order.
+func (c *Core) orderAccelerationParameters(oidB dex.Bytes) (tracker *trackedTrade, swapCoins, accelerationCoins []dex.Bytes, changeCoin dex.Bytes, requiredForRemainingSwaps uint64, err error) {
+	makeError := func(err error) (*trackedTrade, []dex.Bytes, []dex.Bytes, dex.Bytes, uint64, error) {
+		return nil, nil, nil, nil, 0, err
+	}
+
+	if len(oidB) != order.OrderIDSize {
+		return makeError(fmt.Errorf("wrong order ID length. wanted %d, got %d", order.OrderIDSize, len(oidB)))
+	}
+	var oid order.OrderID
+	copy(oid[:], oidB)
```

Comment:
Maybe all the interface methods should just take the `order.OrderID` type rather than `dex.Bytes`? Seems more correct than checking the length at this stage.

Reply:
Generally all the core methods take order id as a `dex.Bytes`, but I will this into the exported core methods.